www/js/lib/zimfile.js

```diff
@@ -110,6 +110,42 @@ define(['xzdec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry'], function(xz, util,
         }
     };
 
+    /**
+     * Reads the whole MIME Type list and returns it as a populated Map
+     * 
+     * @return {Promise} A promise for the MIME Type list as a Map
+     */
+    ZIMFile.prototype.mimeTypeMap = function() {
+        var typeMap = this.mimeTypes;
+        // If we have already populated mimeTypes, we can just return it
+        if (typeMap.size) {
+            return Q.resolve().then(function() {
```

Comment:
Because `ZIMArchive.prototype.getMimetypeString` expects a promise, the result is here wrapped in a promise even though we could get it synchronously. Am I adding unnecessary overhead? It makes for a neat getter function in `zimArchive.js`, but maybe there's a faster way to return results synchronously, with a different architecture in `ZIMArchive.prototype`?